libCacheSim/cache/eviction/LIRS.c

```diff
@@ -210,6 +210,10 @@ static cache_obj_t *LIRS_find(cache_t *cache, const request_t *req,
   obj_s = params->LRU_s->find(params->LRU_s, req, update_cache);
   obj_q = params->LRU_q->find(params->LRU_q, req, update_cache);
 
+  if (update_cache == false) {
```

Comment:
`obj_s != NULL` does not indicate a cache hit. The find function assumes that if we return a `cache_obj`, it is a cache hit.

Reply:
I think we need to check `update_cache` option in the rest of the function